contentcuration/automation/utils/appnexus/base.py

```diff
@@ -20,8 +20,8 @@ def connect(self) -> None:
         pass
 
     @abstractmethod
-    def make_request(self, url: str, params=None) -> Union[bytes, str, Dict]:
-        """ Makes an HTTP request to a given URL using the specified method. """
+    def make_request(self, request) -> Union[bytes, str, Dict]:
```

Comment:
```suggestion
    def make_request(self, request: Request) -> request.Response:
```
If we are going to use `requests.Request` object to make request, in the base class should we add this?

and here:
```
def request(self) -> Request: 
```
as this method is used to create requests.Requests object. and
```
def response(self) -> Union[bytes, str, Dict] 
```
this will return response (Json, Str, Binary)

Reply:
What do you think @akolson @ozer550?

Reply:
Hi @akash5100! Please refer to our discussion on slack. The object we are referring to here is the POJO equivalent in Java but this time in python. Also, we can alter the return type of the abstract method in our implementations to return whatever we want.

Reply:
We'll clean this up so we can return either an `object` or `array` + any other that we anticipate being returned by the backends

Reply:
Alternatively, we can create base backend objects(Which i think is better)from which we can create the other specific backend objects.